pkg/engine/lifecycletest/create_if_not_exists_test.go

```diff
@@ -0,0 +1,883 @@
+package lifecycletest
+
+import (
+	"testing"
+
+	"github.com/blang/semver"
+	. "github.com/pulumi/pulumi/pkg/v3/engine" //nolint:revive
+	"github.com/pulumi/pulumi/pkg/v3/resource/deploy/deploytest"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/plugin"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateIfNotExists_WithImport(t *testing.T) {
+	t.Parallel()
+
+	p := &TestPlan{}
+	project := p.GetProject()
+
+	loaders := []*deploytest.ProviderLoader{
+		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{
+				CreateF: func(
+					urn resource.URN,
+					inputs resource.PropertyMap,
+					timeout float64,
+					preview bool,
+				) (resource.ID, resource.PropertyMap, resource.Status, error) {
+					return "id/a", resource.PropertyMap{}, resource.StatusOK, nil
+				},
+				ReadF: func(
+					urn resource.URN, id resource.ID, inputs, state resource.PropertyMap,
+				) (plugin.ReadResult, resource.Status, error) {
+					return plugin.ReadResult{}, resource.StatusOK, nil
+				},
+			}, nil
+		}),
+	}
+
+	programF := deploytest.NewLanguageRuntimeF(func(_ plugin.RunInfo, monitor *deploytest.ResourceMonitor) error {
+		_, err := monitor.RegisterResource("pkgA:m:typA", "a", true, deploytest.ResourceOptions{
+			ImportID:          "id/a",
+			CreateIfNotExists: "id/a",
+		})
+
+		return err
+	})
+
+	hostF := deploytest.NewPluginHostF(nil, nil, programF, loaders...)
+	options := TestUpdateOptions{
+		T:     t,
+		HostF: hostF,
+	}
+
+	snap, err := TestOp(Update).
+		RunStep(project, p.GetTarget(t, nil), options, false, p.BackendClient, nil, "0")
+
+	assert.Nil(t, snap.VerifyIntegrity(), "Snapshot should be valid")
```

Comment:
```suggestion
	assert.NoError(t, snap.VerifyIntegrity(), "Snapshot should be valid")
```

But also `RunStep` already takes care of snapshot verification, so we don't need to duplicate that here.